resources/ext/config/conf.d/puppetserver.conf

```diff
@@ -1,14 +1,20 @@
+# Where the puppet-agent dependency places puppet, facter, etc...
+# Puppet server expects to load Puppet from this location
+os-settings: {
+    ruby-load-path: [/opt/puppetlabs/puppet/lib/ruby/vendor_ruby]
+}
+
 # configuration for the JRuby interpreters
 jruby-puppet: {
     # This setting determines where JRuby will look for gems.  It is also
     # used by the `puppetserver gem` command line tool.
-    gem-home: /var/lib/puppet/jruby-gems
+    gem-home: /opt/puppetlabs/server/data/puppetserver/jruby-gems
 
     # (optional) path to puppet conf dir; if not specified, will use the puppet default
-    master-conf-dir: /etc/puppetlabs/agent
+    master-conf-dir: /etc/puppetlabs/puppet
 
     # (optional) path to puppet var dir; if not specified, will use the puppet default
-    master-var-dir: /opt/puppetlabs/agent/cache
+    master-var-dir: /opt/puppetlabs/server/data/puppetserver
```

Comment:
> Will `puppet master --configprint vardir` display the value above or fall back to the agent's vardir?

I think as this PR currently stands, you'd still get "/opt/puppetlabs/server/data/puppetserver" after the initial package install because of the following line which would be added to the postinstall in "resources/ext/ezbake.conf":

```
/opt/puppetlabs/puppet/bin/puppet config set --section master vardir  /opt/puppetlabs/server/data/puppetserver
```

Although I'm starting to think maybe that's something we want to avoid doing anyway.  Like if the user had changed the puppet.conf to something else and were doing an upgrade installation, would this automatically stomp whatever the user had set?  And to @cprice404's point, the effective value that the master uses for vardir under puppetserver is sourced from the puppetserver.conf anyway, so the corresponding value in "puppet.conf" seems like it would be irrelevant under puppetserver.

I'd be :+1: on not putting more effort into ensuring that the vardir split is sound until we have a better understanding of where we stand overall around the Puppet 4 AIO + Puppet Server 2 changes in CI - unless there's something @jeffmccune or @joshcooper might know of there that would block our ability to initially get everything to green...